air/src/chiplets/bitwise/mod.rs

```diff
@@ -411,10 +388,6 @@ impl<E: FieldElement> EvaluationFrameExt<E> for &EvaluationFrame<E> {
         binary_not(self.selector(0)) * binary_not(self.selector(1))
     }
     #[inline(always)]
-    fn bitwise_or_flag(&self) -> E {
-        binary_not(self.selector(0)) * self.selector(1)
-    }
-    #[inline(always)]
```

Comment:
1. Thanks. Done.
2. Thanks. Done.
3. Thanks. Done.
4. I did change it for memory as well. It seems like it is including external selectors for bitwise and hasher but without external selectors for memory. So, changed degree for bitwise from 6 to 5 and for memory from 6 to 9. Please let me know if I got it wrong.
5. I've skipped it for now. I'll just check if there are other instances where similar checks need to be made and the best way to achieve it and will either create an issue or submit a PR.

Reply:
Thanks! A few things

Regarding 4 - 
1. In the [chiplets doc file](https://github.com/maticnetwork/miden/blob/f0364f5efa018c1bcac04bf2b57cf45f77fbed00/docs/src/design/chiplets/main.md#chiplets-order) in the table with the chiplet degrees, I think we should have one column for selector degree and one column for internal degree. This will be clearer than the current version and resolve the inconsistency you noted so it doesn't happen again
2. The internal degree of the hash chiplet is 8 (7 + 1 cycle), and the selector is degree 1 (so the total is 9, rather than 8)

Reply:
Regarding 5 -

It's a good engineering practice to write a test whenever a bug is found and submit the fix along with the test that validates it. This means that our test suite will get more robust over time and we'll be less likely to repeat mistakes. Therefore, I think it's important to have something that addresses the bug with the selectors from the first version of this PR and to include that now rather than later.

We can write a test of the output aggregation and ensure that the prover can't set an incorrect value in the output column. It would be great to generalize this, but at a minimum we can use the following test:

```

/// Tests that the prover cannot set an incorrect output during BITWISE_AND at the first row in the
/// cycle when the low limb of a is one.
#[test]
fn output_aggregation_and() {
    let cycle_row = 0;

    // create a valid test frame manually
    let mut current = vec![Felt::ZERO; TRACE_WIDTH];
    let mut next = vec![Felt::ZERO; TRACE_WIDTH];

    let current_bitwise = [
        // selector
        BITWISE_AND[0],
        // a
        Felt::ONE,
        // b
        Felt::new(9),
        // decomposition of a
        Felt::ONE,
        Felt::ZERO,
        Felt::ZERO,
        Felt::ZERO,
        // decomposition of b
        Felt::ONE,
        Felt::ZERO,
        Felt::ZERO,
        Felt::ONE,
        // previous output
        Felt::ZERO,
        // assert a false output
        Felt::new(1337),
    ];

    let next_bitwise = [
        // selector
        BITWISE_AND[0],
        // a
        Felt::new(19),
        // b
        Felt::new(157),
        // decomposition of a
        Felt::ONE,
        Felt::ONE,
        Felt::ZERO,
        Felt::ZERO,
        // decomposition of b
        Felt::ONE,
        Felt::ZERO,
        Felt::ONE,
        Felt::ONE,
        // previous output
        Felt::new(1337),
        // output
        Felt::new(21393),
    ];

    current[BITWISE_TRACE_RANGE].copy_from_slice(&current_bitwise);
    next[BITWISE_TRACE_RANGE].copy_from_slice(&next_bitwise);

    let frame = EvaluationFrame::<Felt>::from_rows(current, next);
    let result = get_constraint_evaluation(frame, cycle_row);

    // expect a failure for the output aggregation constraint (the last one)
    assert_ne!(Felt::ZERO, result[NUM_CONSTRAINTS - 1]);
}
```

Reply:
Thank you @grjte.
Re 4 -
I've updated the degrees in the docs and the diagram with what you suggested.
Re 5 -
Thank you for the explanation for the failing test. I think I misunderstood your comment earlier. I've added the failure test you suggested and will keep this in mind in the future.